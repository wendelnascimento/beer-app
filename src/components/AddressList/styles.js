import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const List = styled.ul`
  list-style-type: none;
  padding: 16px 0;
  margin: 0;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
  border-top: 1px solid #eee;
  box-shadow: rgba(0, 0, 0, 0.6) 0px 14px 23px -16px;
  background: #ffffff;
  max-height: 300px;
  overflow-y: scroll;
`;

export const ListItem = styled.li`
  padding: 8px 24px;
  width: 100%;

  &:not(:last-child) {
    border-bottom: 1px solid #fafafa;
    padding-bottom: 24px;
  }
`;

export const ItemTitle = styled.span`
  display: block;
  font-size: 1.2rem;
  color: #000000;
`;

export const ItemVicinity = styled.span`
  display: block;
  color: #cccccc;
  font-size: 1rem;
`;

export const CustomLink = styled(Link)`
  text-decoration: none;
`;
