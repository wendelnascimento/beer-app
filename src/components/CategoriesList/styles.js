import styled from 'styled-components';

export const List = styled.ul`
  list-style-type: none;
  padding: 0;
  margin: 0 0 16px;
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  grid-gap: 16px;

  @media screen and (max-width: 900px) {
    grid-template-columns: repeat(5, 1fr);
  }

  @media screen and (max-width: 600px) {
    grid-template-columns: repeat(4, 1fr);
  }

  @media screen and (max-width: 460px) {
    grid-template-columns: repeat(3, 1fr);
  }

  @media screen and (max-width: 350px) {
    grid-template-columns: repeat(2, 1fr);
  }
`;

export const ListItem = styled.li`
  border: 2px solid #000;
  border-radius: 5px;
  overflow: hidden;
  transition: border-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);

  &:hover {
    border-color: #f28e1c;
  }
`;

export const ItemButton = styled.button`
  width: 100%;
  padding: 8px 16px;
  border: none;
  font-weight: 700;
  text-align: center;
  background: #ffffff;

  &:hover {
    cursor: pointer;
  }
`;
