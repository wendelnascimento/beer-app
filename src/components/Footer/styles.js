import styled from 'styled-components';

export const Wrapper = styled.footer`
  padding: 16px;
  color: #cccccc;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  border-top: 1px solid #fafafa;
`;
