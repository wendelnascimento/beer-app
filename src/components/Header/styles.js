import styled from 'styled-components';

export const Wrapper = styled.header`
  max-width: 100vw;
  background: #fff;
  padding: 16px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  border-bottom: 1px solid #fafafa;
`;

export const Title = styled.h1`
  font-weight: 300;
  margin: 0;
`;
