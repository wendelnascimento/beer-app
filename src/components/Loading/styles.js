import styled, { keyframes } from 'styled-components';

export const bounceAndRotate = keyframes`
  0% {
    transform: scale3d(.8, .8, 1);
  }
  25% {
    transform: scale3d(1.5, 1.5, 1);
  }
  50% {
    transform: scale3d(.8, .8, 1);
  }
  75% {
    transform: rotate3d(0, 0, 1, 180deg);
  }
  100% {
    transform: scale3d(1, 1, 1), rotate3d(0, 0, 1, 360deg);
  }
`;

export const Wrapper = styled.div`
  position: fixed;
  z-index: 10;
  width: 100vw;
  height: 100vh;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgba(255, 255, 255, 0.6);
`;

export const Beer = styled.span`
  font-size: 7rem;
  transform: scale3d(1, 1, 1), rotate3d(0, 0, 1, 0deg);
  animation: ${bounceAndRotate} 2s infinite linear;
`;
