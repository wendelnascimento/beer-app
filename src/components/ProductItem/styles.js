import styled from 'styled-components';

export const Card = styled.li`
  width: 100%;
  max-width: 300px;
  background: #ffffff;
  border: 2px solid #fff;
  display: flex;
  flex-direction: column;
  border-radius: 5px;
  box-shadow: rgba(0, 0, 0, 0.6) 0px 2px 15px -5px;
  transform: translate3d(0, 0, 0);
  transition: transform 0.3s cubic-bezier(0.19, 1, 0.22, 1),
    border-color 0.3s cubic-bezier(0.19, 1, 0.22, 1);
  overflow: hidden;

  &:hover {
    transform: translate3d(0, -5px, 0);
    border-color: #f28e1c;
  }
`;

export const ProductImage = styled.img`
  width: 100%;
  object-fit: scale-down;
  background: #ffffff;
  border-radius: 5px;
  margin-bottom: 8px;
`;

export const ImagePlaceholder = styled.span`
  font-size: 5rem;
  text-align: center;
  margin-top: auto;
  margin-bottom: auto;
`;

export const ProductTitle = styled.span`
  color: #666;
  font-size: 0.8rem;
  margin-top: auto;
  padding: 0 8px;
`;

export const ProductPrice = styled.span`
  color: #f28e1c;
  font-size: 1.2rem;
  font-weight: 700;
  margin-bottom: 16px;
  padding: 0 8px;
`;

export const QuantityRow = styled.div`
  align-self: stretch;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const AddButton = styled.button`
  width: 30px;
  height: 30px;
  background: #f28e1c;
  color: #ffffff;
  border: none;
  font-size: 1.3rem;
  font-weight: 300;

  &:hover {
    cursor: pointer;
  }
`;

export const RemoveButton = styled(AddButton)`
  background: #b03c3c;
`;
