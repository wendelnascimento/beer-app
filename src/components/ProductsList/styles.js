import styled from 'styled-components';

export const List = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0 32px 32px;
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-gap: 16px 32px;
  text-align: center;

  @media screen and (max-width: 900px) {
    grid-template-columns: repeat(3, 1fr);
  }

  @media screen and (max-width: 700px) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media screen and (max-width: 390px) {
    grid-template-columns: repeat(1, 1fr);
  }
`;

export const EmptyText = styled.h2`
  margin-left: auto;
  margin-right: auto;
`;
