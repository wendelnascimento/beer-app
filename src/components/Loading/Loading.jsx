import React from 'react';

import { Wrapper, Beer } from './styles';

const Loading = () => (
  <Wrapper>
    <Beer>
      <span role="img" aria-label="Beer App">
        🍺
      </span>
    </Beer>
  </Wrapper>
);

export default Loading;
