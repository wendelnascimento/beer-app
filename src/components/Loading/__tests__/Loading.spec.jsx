import React from 'react';
import { shallow, mount } from 'enzyme';
import { css } from 'styled-components';

import Loading from '../Loading';

import { bounceAndRotate, Beer } from '../styles';

describe('<Loading />', () => {
  it('should render', () => {
    shallow(<Loading />);
  });

  it('should match stored snapshot', () => {
    const wrapper = mount(<Loading />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should be animating', () => {
    const wrapper = mount(<Loading />);

    const keyframesObject = css`
      ${bounceAndRotate}
    `;
    const animation = `${keyframesObject[1].name} 2s infinite linear`;

    expect(wrapper.find(Beer)).toHaveStyleRule('animation', animation);
  });
});
