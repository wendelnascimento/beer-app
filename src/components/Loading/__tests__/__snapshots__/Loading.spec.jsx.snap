// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should match stored snapshot 1`] = `
.c0 {
  position: fixed;
  z-index: 10;
  width: 100vw;
  height: 100vh;
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(255,255,255,0.6);
}

.c1 {
  font-size: 7rem;
  -webkit-transform: scale3d(1,1,1),rotate3d(0,0,1,0deg);
  -ms-transform: scale3d(1,1,1),rotate3d(0,0,1,0deg);
  transform: scale3d(1,1,1),rotate3d(0,0,1,0deg);
  -webkit-animation: bFhvm 2s infinite linear;
  animation: bFhvm 2s infinite linear;
}

<Loading>
  <styled.div>
    <div
      className="c0"
    >
      <styled.span>
        <span
          className="c1"
        >
          <span
            aria-label="Beer App"
            role="img"
          >
            üç∫
          </span>
        </span>
      </styled.span>
    </div>
  </styled.div>
</Loading>
`;
